mobius-rx/src/test/java/com/spotify/mobius/rx/TransformersTest.java

```diff
@@ -100,4 +101,33 @@ public void consumerTransformerShouldPropagateCompletion() throws Exception {
     subscriber.awaitTerminalEvent(1, TimeUnit.SECONDS);
     subscriber.assertCompleted();
   }
+
+  @Test
+  public void effectPerformerInvokesFunctionWithReceivedEffectAndEmitsReturnedEvents() {
+    PublishSubject<String> upstream = PublishSubject.create();
+    TestScheduler scheduler = new TestScheduler();
+    Function<String, Integer> function = s -> s.length();
+    AssertableSubscriber<Integer> observer =
+        upstream.compose(Transformers.fromFunction(function, scheduler)).test();
+
+    upstream.onNext("Hello");
+    scheduler.triggerActions();
+    observer.assertValue(5);
+  }
+
+  @Test
+  public void effectPerformerInvokesFunctionWithReceivedEffectAndErrorsForUnhandledExceptions() {
```

Comment:
There should also be a test validating whether it is possible to continue processing events submitted to the original observable, I think. See comment about `Transformers.fromFunction` javadocs.

Reply:
As discussed, functions throwing exceptions will lead to irrecoverable error that will render this transformer useless